sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -486,14 +531,28 @@ case class FlatMapGroupsWithState(
     outputMode: OutputMode,
     isMapGroupsWithState: Boolean = false,
     timeout: GroupStateTimeout,
-    child: LogicalPlan) extends UnaryNode with ObjectProducer {
+    hasInitialState: Boolean = false,
+    initialStateGroupAttrs: Seq[Attribute] = Seq.empty,
+    initialStateEncoder: ExpressionEncoder[Any],
+    initialState: LogicalPlan,
+    child: LogicalPlan) extends BinaryNode with ObjectProducer {
 
   if (isMapGroupsWithState) {
     assert(outputMode == OutputMode.Update)
   }
 
-  override protected def withNewChildInternal(newChild: LogicalPlan): FlatMapGroupsWithState =
-    copy(child = newChild)
+  if (hasInitialState && !child.isStreaming) {
```

Comment:
Any reason we don't check this in UnsupportedOperationChecker.checkForBatch?

Reply:
fixed